code/datums/components/radioactive.dm

```diff
@@ -0,0 +1,125 @@
+/// If a thing is to be radioactive, slap this component on it. Only call mob.take_radiation_dose directly as a last resort.
+TYPEINFO(/datum/component/radioactive)
+	initialization_args = list(
+		ARG_INFO("radStrength", DATA_INPUT_NUM, "Value of radiation strength \[0-100\]", 100),
+		ARG_INFO("decays", DATA_INPUT_BOOL, "Whether this radiation will decay over time (bool)", FALSE),
+		ARG_INFO("neutron", DATA_INPUT_BOOL, "Whether this radiation is neutron, and therefor penetrates more (bool)", FALSE),
+		ARG_INFO("effectRange", DATA_INPUT_NUM, "How far this effect goes. Do not set too high, it's expensive. \[0-10\]", 1)
+	)
+
+/datum/component/radioactive
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/radStrength = 0
+	var/decays = FALSE
+	var/neutron = FALSE
+	var/_added_to_items_processing = FALSE
+	var/effect_range = 1
+	var/backup_color = null //so hacky
+
+	Initialize(radStrength=100, decays=FALSE, neutron=FALSE, effectRange=1)
+		if(!istype(parent,/atom))
+			return COMPONENT_INCOMPATIBLE
+		. = ..()
+		src.radStrength = radStrength
+		src.decays = decays
+		src.neutron = neutron
+		src.effect_range = effectRange
+		if(parent.GetComponent(src.type)) //don't redo the filters and stuff if we're a duplicate
+			return
+		RegisterSignal(parent, list(COMSIG_ATOM_EXAMINE), .proc/examined)
+		RegisterSignal(parent, list(COMSIG_ATOM_CROSSED,
+			COMSIG_ATOM_ENTERED,
+			COMSIG_ATTACKHAND,
+			COMSIG_ITEM_EQUIPPED,
+			COMSIG_ITEM_PICKUP,
+			COMSIG_MOB_GRABBED,
+			COMSIG_ITEM_ATTACK_POST,
+		), .proc/touched)
+		RegisterSignal(parent, list(COMSIG_ITEM_CONSUMED_PARTIAL, COMSIG_ITEM_CONSUMED_ALL), .proc/eaten)
+
+		if(isitem(parent))
+			RegisterSignal(parent, list(COMSIG_ITEM_PROCESS), .proc/ticked)
+			if(!(parent in global.processing_items))
+				global.processing_items.Add(parent)
+				src._added_to_items_processing = TRUE
+		else if(ismob(parent))
+			RegisterSignal(parent, list(COMSIG_LIVING_LIFE_TICK), .proc/ticked)
+		else
+			global.processing_items.Add(src) //gross - in the event that this component is put on something that isn't an item/mob, use the item processing loop anyway
```

Comment:
essentially because `COMSIG_ITEM_PROCESS` only gets sent to items registered in `global.processing_items`, and those items need to implement `process()` anyway. I suppose I could register the `COMSIG_ITEM_PROCESS` signal and leave the `process()` as a stub instead, but it's functionally equivalent.